packages/api/amplify_api/lib/src/api_plugin_impl.dart

```diff
@@ -0,0 +1,397 @@
+  /// OIDC or Lambda).
+  ///
+  /// If an endpoint has an API key, ensure valid auth provider registered.
+  ///
+  /// Register OIDC/Lambda set to _authProviders in constructor.
+  void _registerApiPluginAuthProviders() {
+    _apiConfig.endpoints.forEach((key, value) {
+      // Check the presence of apiKey (not auth type) because other modes might
+      // have a key if not the primary auth mode.
+      if (value.apiKey != null) {
+        _authProviderRepo.registerAuthProvider(
+          APIAuthorizationType.apiKey.authProviderToken,
+          AppSyncApiKeyAuthProvider(),
+        );
+      }
+    });
+
+    // Register OIDC/Lambda auth providers.
+    for (final authProvider in _authProviders.values) {
+      _authProviderRepo.registerAuthProvider(
+        authProvider.type.authProviderToken,
+        OidcFunctionAuthProvider(authProvider),
+      );
+    }
+  }
+
+  @override
+  Future<void> addPlugin({
+    required AmplifyAuthProviderRepository authProviderRepo,
+  }) async {
+    await super.addPlugin(authProviderRepo: authProviderRepo);
+
+    if (zIsWeb || !(Platform.isAndroid || Platform.isIOS)) {
+      return;
+    }
+
+    // Configure this plugin to act as a native iOS/Android plugin.
+    final nativePlugin = _NativeAmplifyApi(_authProviders);
+    NativeApiPlugin.setup(nativePlugin);
+
+    final nativeBridge = NativeApiBridge();
+    try {
+      final authProvidersList =
+          _authProviders.keys.map((key) => key.rawValue).toList();
+      await nativeBridge.addPlugin(authProvidersList);
+    } on PlatformException catch (e) {
+      if (e.code == 'AmplifyAlreadyConfiguredException' ||
+          e.code == 'AlreadyConfiguredException') {
+        throw const AmplifyAlreadyConfiguredException(
+          AmplifyExceptionMessages.alreadyConfiguredDefaultMessage,
+          recoverySuggestion:
+              AmplifyExceptionMessages.alreadyConfiguredDefaultSuggestion,
+        );
+      }
+      throw AmplifyException.fromMap((e.details as Map).cast());
+    }
+  }
+
+  /// Returns the HTTP client to be used for REST/GraphQL operations.
+  ///
+  /// Use [apiName] if there are multiple endpoints of the same type.
+  @visibleForTesting
+  AWSHttpClient getHttpClient(
+    EndpointType type, {
+    String? apiName,
+    APIAuthorizationType? authorizationMode,
+  }) {
+    final endpoint = _apiConfig.getEndpoint(
+      type: type,
+      apiName: apiName,
+    );
+    final authModeForClientKey =
+        authorizationMode ?? endpoint.config.authorizationType;
+    final clientPoolKey = '${endpoint.name}.${authModeForClientKey.name}';
+    return _clientPool[clientPoolKey] ??= AmplifyHttpClient(
+      baseClient: AmplifyAuthorizationRestClient(
+        endpointConfig: endpoint.config,
+        baseClient: _baseHttpClient,
+        authorizationMode: authorizationMode,
+        authProviderRepo: _authProviderRepo,
+      ),
+    )..supportedProtocols = SupportedProtocols.http1;
```

Comment:
Can this be moved to the constructor?

Reply:
Do you mean making it an option of the client here? Or setting on the base client in constructor for this client?